### PR TITLE
(#891) findbugs should be enabled in qulice

### PR DESCRIPTION
**Pull request by ilyakharlamov**
This PR:
Enables FindBugs (#891)  - 
Since external dependencies [are not allowed](https://github.com/yegor256/cactoos/pull/1020#issuecomment-457833335), the exclusion of the check can only be done on per-class basis in pom.xml

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1044](https://github.com/yegor256/cactoos/issues/1044) is now in scope, role is `REV`

File: pom.xml [hunk 1]
```diff
@@ -138,7 +138,18 @@ The MIT License (MIT)
             <configuration>
               <excludes combine.children="append">
                 <exclude>checkstyle:/src/site/resources/.*</exclude>
-                <exclude>findbugs:.*</exclude>
+                <exclude>findbugs:org.cactoos.text.TextEnvelopeTest</exclude>
```
Comment (fabriciofx) [inline pom.xml]: @ilyakharlamov Is there anything we can do in the future to remove these exclusions? If yes, WDYT put a PDD what it need to do?

Comment (ilyakharlamov) [reply pom.xml]: @fabriciofx One issue can be fixed if #1028 is merged. I also opened a PR for qulice that will allow exclusions on per-class, per-rule and per-method basis.

Comment (fabriciofx) [reply pom.xml]: @ilyakharlamov Right.

**Comment by fabriciofx** [comment]
@ilyakharlamov This PR seems fine to me. Congratulations!

**Comment by fabriciofx** [comment]
@rultor merge

**Comment by llorllale** [comment]
@ilyakharlamov I expected a puzzle to fix those classes and remove them from the exclusion list

**Comment by ilyakharlamov** [comment]
@llorllale please confirm you'll be OK with the following changes to overcome FindBugs complaints: `ListIterator` to be renamed to `ListIteratorOf` (since it shadows`java.util.ListIterator`)
`make NumberOf serializable` - that will cause the chain reaction to make lots of other scalars serializable
Most of other fixes have to stay as exclusions because it's not possible to fix them: (for example FindBugs complains that org.cactoos.iterator.Endless doesn't throw NoSuchElement, but Endless must never throw NoSuchElement by definition)

**Comment by llorllale** [comment]
@ilyakharlamov I am OK with `ListIteratorOf` but not the rest. I would have liked to teamed/qulice#1020 but I don't want to deal with all the new linting issues that will have to be fixed if we upgrade qulice.

Rename `ListIterator` and let's call it a day.

**Comment by ilyakharlamov** [comment]
@llorllale  renamed, rebased

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Code review was too long (14 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
The job [#1044](https://github.com/yegor256/cactoos/issues/1044) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Pull request merged**